frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -268,6 +268,26 @@ pub type CompactAccuracyOf<T> = <CompactOf<T> as CompactSolution>::Accuracy;
 /// The accuracy of the election, when computed on-chain. Equal to [`Config::OnChainAccuracy`].
 pub type OnChainAccuracyOf<T> = <T as Config>::OnChainAccuracy;
 
+/// A voter's fundamental data: their ID, their stake, and the list of candidates for whom they voted.
+pub type Voter<T> = (
+	<T as frame_system::Config>::AccountId,
+	sp_npos_elections::VoteWeight,
+	Vec<<T as frame_system::Config>::AccountId>,
+);
+
+/// The relative distribution of a voter's stake among the winning targets.
+pub type Assignment<T> = sp_npos_elections::Assignment<
+	<T as frame_system::Config>::AccountId,
+	CompactAccuracyOf<T>,
+>;
+
+// The [`IndexAssignment`] type specialized for a particular runtime `T`.
+pub type IndexAssignmentOf<T> = IndexAssignment<
```

Comment:
ahh you actually have this, I'd move this to `sp-npos` crates to be more reusable.

Reply:
3dda173e4

Ended up leaving a minor type alias here to keep the refactor small; the `IndexAssignmentOf` here is parametrized by the `Runtime`, whereas `sp_npos_elections::IndexAssignmentOf` is parametrized by the `Compact`.